flow/beta/cli/project.go

```diff
@@ -29,59 +29,77 @@ import (
 	"github.com/onflow/flow-cli/flow/beta/cli/keys"
 )
 
+// Project structure containing current config
 type Project struct {
-	conf     *config.Config
-	accounts []*Account
+	Config config.Config
 }
 
-const DefaultConfigPath = "flow.json"
+// Contract structure defines Name of the contract, Source of
+// the contract (path, url...), Target where to deploy contract to (account address)
+type Contract struct {
+	Name   string
+	Source string
+	Target flow.Address
+}
 
+// REF: this should be moved to config as it deals with config
+// LoadProject parses config and create project instance
 func LoadProject() *Project {
-	conf, err := config.Load(DefaultConfigPath)
+	configLocation := "flow.json" //TODO: support this from cli
+
+	conf, err := config.Load(configLocation)
 	if err != nil {
 		if errors.Is(err, config.ErrDoesNotExist) {
 			Exitf(
 				1,
 				"Project config file %s does not exist. Please initialize first\n",
-				DefaultConfigPath,
+				configLocation,
 			)
 		}
 
-		Exitf(1, "Failed to open project configuration in %s", DefaultConfigPath)
+		Exitf(1, "Failed to open project configuration in %s", configLocation)
 
 		return nil
 	}
 
-	proj, err := newProject(conf)
-	if err != nil {
-		// TODO: replace with a more detailed error message
-		Exitf(1, "Invalid project configuration: %s", err)
-	}
-
-	return proj
-}
-
-func ProjectExists() bool {
-	return config.Exists(DefaultConfigPath)
+	return newProject(conf)
 }
 
+//REF: discuss and rethink what project init is and where configuration init should be stored
+// InitProject create project structure
 func InitProject() *Project {
-	serviceAccount, serviceAccountKey := generateEmulatorServiceAccount()
+	serviceAccount := generateEmulatorServiceAccount()
 
+	config := config.Config{
```

Comment:
> discuss and rethink what project init is and where configuration init should be stored

I wasn't sure about the best way to structure this. I ended up defining `defaultConfig` in `project.go` because I was trying to keep the `config` package "stateless" by making it only describe the config _structure_, not the default values.

In general, I tried to decouple "project" and "config" so that it'd be easier to support multiple config formats without having to change the actual project model. For example, there could be multiple backwards-compatible versions of the config schema, or even a YAML version.

However, there ended up being a lot of overlap. I'm fine with moving all of the config-related logic (including default values) to the `config` package. In fact, what if we just combined `project.go` and `config` into a single package? We can try to create a config abstraction later if it's needed.

Reply:
I agree. In general, I think default values should be part of the config but shouldn't be tied to JSON config. I strongly agree with the possibility of having different formats. I think config with value validation and default values etc. should be a layer above that. I would at that point create different config parsers (some would support different formats and others would support older versions) but then all the parsers would return a structure defined in config. Right now since we only support one format all of that is together but I would at a later point just move parsing of JSON to another file but keep the structures I've created in the config file since all further operations are based on those structures and hence provide an abstraction over the concrete format of the config file.

Reply:
> I think config with value validation and default values etc. should be a layer above that.

+1 for separating the config structure from the actual encoding logic. However, I think you're right that we should just focus on the JSON structure for now and split it out later if needed.

>  I want to expose config at a later point as a single point of config for the whole cli and hence should be generic

I agree 👍